llvm-spirv/lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -288,6 +288,49 @@ void SPIRVRegularizeLLVMBase::expandVIDWithSYCLTypeByValComp(Function *F) {
       nullptr, &Attrs, true);
 }
 
+// intel/llvm customization
```

Comment:
Why can't we make it a part of the upstream translator? If we can't, we should have a tracker somewhere listing all our translator customizations, so this PR should be recorded there as a reminder for us to revert it.

What about turning this into a pass which we can run in our device compilation pipeline instead of hacking it into the translator?

Reply:
If/when landed I'll definitely add this PR to https://github.com/intel/llvm/issues/7592

> Why can't we make it a part of the upstream translator?

I'm not sure about this approach. We definitely need to do `something` for matrix unwrapping just for -O0 case, may be a pass in sycl-post-link or in the translator or alternative for both of them is adding some sort of the translator builtin (like __translate_sampler_initializer and others in OpenCL) with the translator inserting the definition of this builtin. I'd like to try unwrapping first without introducing new LLVM IR entity like translator builtin, but want to ensure that such solution is stable.

> What about turning this into a pass which we can run in our device compilation pipeline instead of hacking it into the translator?

I haven't though about it tbh, can move it there.